File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha~conf_via-dynamic_config/csharp/App_Code/ValidationCounter.cs
using System;
using System.Web;

namespace CaptchaRequestDynamicSettingsExampleCSharp
{
    public class ValidationCounter
    {
        const string FailedValidationsCountKey = "FailedValidationsCountKey";

        public static int GetFailedValidationsCount()
        {
            int count = 0;
            object saved = HttpContext.Current.Session[FailedValidationsCountKey];
            if (null != saved)
            {
                try
                {
                    count = (int)saved;
                }
                catch (InvalidCastException)
                {
                    // ignore cast errors
                    count = 0;
                }
            }
            return count;
        }

        public static void IncrementFailedValidationsCount()
        {
            int count = GetFailedValidationsCount();
            count++;
            HttpContext.Current.Session[FailedValidationsCountKey] = count;
        }

        public static void ResetFailedValidationsCount()
        {
            HttpContext.Current.Session.Remove(FailedValidationsCountKey);
        }
    }
}

File: bin/Debug/net5.0/v4.4.2/lgcy-on-lgcy/examples/t_api-captcha~conf_via-dynamic_config/vbnet/App_Code/ValidationCounter.vb
Friend Class ValidationCounter
    Const FailedValidationsCountKey As String = "FailedValidationsCountKey"

    Public Shared Function GetFailedValidationsCount() As Integer

        Dim count As Integer = 0
        Dim saved As Object = HttpContext.Current.Session(FailedValidationsCountKey)
        If (Not (saved Is Nothing)) Then

            Try

                count = DirectCast(saved, Integer)

            Catch ex As InvalidCastException

                ' ignore cast errors
                count = 0
            End Try
        End If
        Return count

    End Function
    Public Shared Sub IncrementFailedValidationsCount()

        Dim count As Integer = GetFailedValidationsCount()
        count += 1
        HttpContext.Current.Session(FailedValidationsCountKey) = count
    End Sub

    Public Shared Sub ResetFailedValidationsCount()

        HttpContext.Current.Session.Remove(FailedValidationsCountKey)
    End Sub
End Class
